spec/requests/api/v1/job_applications/job_application_show_spec.rb

```diff
@@ -0,0 +1,152 @@
+        expect(jobApp[:data][:attributes][:contact_information]).to eq(@facebook_application[:contact_information])
+        expect(jobApp[:data][:attributes][:company_id]).to eq(@facebook_application[:company_id])      
+      end
+    end
+
+    context "sad path" do
+
+      it "returns error serializer if job application id does not exist" do
+
+        post api_v1_sessions_path, params: { email: @user.email, password: "Jolene123" }, as: :json
+        
+        token = JSON.parse(response.body)["token"]
+        
+        get "/api/v1/users/#{@user.id}/job_applications/0", 
+        headers: {"Authorization" => "Bearer #{token}" }, as: :json
+
+        expect(response).to_not be_successful
+        expect(response.status).to eq(404)
+
+        json = JSON.parse(response.body, symbolize_names: true)
+
+
+        expect(json[:message]).to eq("Job application not found")
+        expect(json[:status]).to eq(404)
+      end
+
+      it "returns error serializer if job application id belongs to another user" do
+        user_2 = User.create!(name: "Daniel Averdaniel", email: "daderdaniel@gmail.com", password: "nuggetonnabiscut")
+  
+        user_2_application = JobApplication.create!(
+          position_title: "Crank Operator",
+          date_applied: "2024-10-31",
+          status: 1,
+          notes: "Not sure im familiar with the tech-stack",
+          job_description: "You turn the big crank that powers google",
+          application_url: "www.example.com",
+          contact_information: "owneroperator@gmail.com",
+          company_id: @google.id,
+          user_id: user_2.id
+        )
+
+        post api_v1_sessions_path, params: { email: @user.email, password: "Jolene123" }, as: :json
+        
+        token = JSON.parse(response.body)["token"]
+
+        get "/api/v1/users/#{@user.id}/job_applications/#{user_2_application.id}",  
+        headers: {"Authorization" => "Bearer #{token}" }, as: :json
+
+        expect(response).to_not be_successful
+        expect(response.status).to eq(404)
+
+        json = JSON.parse(response.body, symbolize_names: true)
+
+        expect(json[:message]).to eq("Job application not found")
+        expect(json[:status]).to eq(404)
+      end
+
+      # it "returns error serializer if no params are passed in URL for job application id" do
+
+      #   post api_v1_sessions_path, params: { email: @user.email, password: "Jolene123" }, as: :json
+        
+      #   token = JSON.parse(response.body)["token"]
+
+      #    params = nil
+
+      #   get "/api/v1/users/#{@user.id}/job_applications/#{params}",
+      #   # params: {id: nil},
+      #   headers: {"Authorization" => "Bearer #{token}" }, as: :json
+
+      #   expect(response).to_not be_successful
+      #   expect(response.status).to eq(400)
+
+      #   json = JSON.parse(response.body, symbolize_names: true)
+
+      #   expect(json[:message]).to eq("Job application ID is missing")
+      #   expect(json[:status]).to eq(400)
+      # end
```

Comment:
We have worked together on this sad path test in attempts to get it up and running. I hope we can revisit it in the future, as I believe it’s an important edge case to cover. With the recent addition of authorizations, we encountered an issue related to the Pundit gem and the show action route.